lib/clowne/utils/operation.rb

```diff
@@ -35,20 +35,27 @@ def clear!
       attr_reader :mapper
 
       def initialize(mapper)
-        @blocks = []
+        @after_clone_blocks = []
+        @after_persist_blocks = []
         @mapper = mapper
       end
 
       def add_after_persist(block)
-        @blocks.unshift(block)
+        @after_persist_blocks.unshift(block)
+      end
+
+      def add_after_clone(block)
+        @after_clone_blocks.unshift(block)
       end
 
       def add_mapping(origin, clone)
         @mapper.add(origin, clone)
       end
 
       def to_record
-        @clone
+        @clone.tap do
+          run_after_clone
```

Comment:
Looks like It will be executed many times:
```ruby
ACloner < Clowne::Cloner
  after_clone do |*|
    puts "I'm calling you"
  end
end

operation = ACloner.call(record)
operation.to_record
# => "I'm calling you"
operation.persist
# => "I'm calling you"
3.times { operation.to_record }
# => "I'm calling you"
# => "I'm calling you"
# => "I'm calling you"
```

If so, we should call `after_clone` only once. (like you have already implemented for Sequel)

Reply:
You are right, it was called multiple times. I added result memoization to Clowne::Utils::Operation#to_record, now `after_clone` is called only once.

Reply:
Nice! Just one last remark - what do you think about avoiding of using `tap` (because it is redundant):
```ruby
def to_record
  return @_record if defined?(@_record)
  run_after_clone
  @_record = @clone
end
```

Or `tap` because of Rubocop stricts?

Reply:
Thank you for remark! I removed redundant `tap` call, now it looks better.